Port MASTG-TEST-0082: Testing whether the App is Debuggable (ios) (by @appknox) (#3097)
OWASP/owasp-mastg: tests-beta/ios/MASVS-RESILIENCE/MASTG-TEST-0x82.md

---
platform: ios

> title: Testing Whether the App is Debuggable

Reviewer: ```suggestion
title: Debuggable Entitlement Enabled in the entitlements.plist
```

id: MASTG-TEST-0x82
type: [static, dynamic]

> weakness: MASWE-0101

Reviewer: MASWE-0067: Debuggable Flag Not Disabled

```suggestion
weakness: MASWE-0067
```

---
platform: ios
title: Testing Whether the App is Debuggable
id: MASTG-TEST-0x82
type: [static, dynamic]
weakness: MASWE-0101
---

## Overview

> The test evaluates whether an iOS application is configured to allow debugging. If an app is debuggable, attackers can leverage debugging tools to reverse-engineer the application, analyse its runtime behaviour, and potentially compromise sensitive data or functionality.

Reviewer: ```suggestion
The test evaluates whether an iOS application is configured to allow debugging. If an app is debuggable, attackers can leverage debugging tools (see @MASTG-TECH-0084) to analyse the runtime behaviour of the app, and potentially compromise sensitive data or functionality.
```

## Steps

1. Run a static analysis using @MASTG-TOOL-0111 to extract entitlements from the binary to check the value of the `get-task-allow` key and is set to `true`.
2. Run a [dynamic analysis](../../../techniques/ios/MASTG-TECH-0084.md) using @MASTG-TOOL-0057.

## Observation

The entitlement get-task-allow is false, and anti-reverse engineering measures prevent debugger attachment attempts.

## Evaluation

> The test fails as the entitlement get-task-allow is true, allowing debugger attachment.

Reviewer: ```suggestion
The test fails if the `get-task-allow` entitlement is `true`.
```
